pkg/scheduler/event_handler.go

```diff
@@ -229,26 +229,21 @@ func (s *Scheduler) updateCluster(oldObj, newObj interface{}) {
 	case !equality.Semantic.DeepEqual(oldCluster.Labels, newCluster.Labels):
 		fallthrough
 	case !equality.Semantic.DeepEqual(oldCluster.Spec, newCluster.Spec):
-		s.enqueueAffectedPolicy(newCluster)
-		s.enqueueAffectedClusterPolicy(newCluster)
+		s.enqueueAffectedPolicy(oldCluster, newCluster)
+		s.enqueueAffectedClusterPolicy(oldCluster, newCluster)
 	}
 }
 
 // enqueueAffectedPolicy find all propagation policies related to the cluster and reschedule the RBs
-func (s *Scheduler) enqueueAffectedPolicy(newCluster *clusterv1alpha1.Cluster) {
+func (s *Scheduler) enqueueAffectedPolicy(oldCluster, newCluster *clusterv1alpha1.Cluster) {
 	policies, _ := s.policyLister.List(labels.Everything())
 	for _, policy := range policies {
 		selector := labels.SelectorFromSet(labels.Set{
 			policyv1alpha1.PropagationPolicyNamespaceLabel: policy.Namespace,
 			policyv1alpha1.PropagationPolicyNameLabel:      policy.Name,
 		})
 		affinity := policy.Spec.Placement.ClusterAffinity
-		switch {
-		case affinity == nil:
-			// If no clusters specified, add it in queue
-			fallthrough
-		case util.ClusterMatches(newCluster, *affinity):
-			// If specific cluster matches the affinity. add it in queue
+		if affinity == nil || util.ClusterMatches(oldCluster, *affinity) || util.ClusterMatches(newCluster, *affinity) {
```

Comment:
1. I think we need to add some comments.
2. I think we should match the newCluster first, then the oldCluster.

Reply:
PTAL